auto_round/script/llm.py

```diff
@@ -534,6 +536,33 @@ def tune(args):
 
     enable_torch_compile = True if "--enable_torch_compile" in sys.argv else False
 
+    # sq
+    if args.sq:
+        from auto_round.calib_dataset import get_dataloader
+
+        dataloader = get_dataloader(tokenizer, args.seqlen, bs=8, nsamples=args.nsamples)
+        auto_alpha_args = {
+            "init_alpha": 0.5,
+            "alpha_min": 0.1,
+            "alpha_max": 1.0,
+            "alpha_step": 0.1,
+            "shared_criterion": "mean",
+            "n_samples": 512,  ##512 for cuda, 128 for cpu?
+            # "do_blockwise": True
+        }
+        from auto_round.smooth_quant import SmoothQuant
+
+        model = model.to(device_str)
+        sq = SmoothQuant(model, dataloader, device=model.device, group_size=-1)
+        model = sq.transform_model(
+            alpha=0.5,
+            # alpha="auto",
```

Comment:
when using auto alpha, layer_config and all the quantization config like data type must know